src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -315,31 +437,94 @@ struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
             // If Find doesn't return CHIP_NO_ERROR, the scene wasn't found, which doesn't return an error
             VerifyOrReturnValue(this->Find(scene_id, scene.index) == CHIP_NO_ERROR, CHIP_NO_ERROR);
 
+            // Update the global scene count
+            GlobalSceneCount global_scene_count;
+            ReturnErrorOnFailure(global_scene_count.Load(storage));
+            global_scene_count.count_value--;
+            ReturnErrorOnFailure(global_scene_count.Save(storage));
+
             scene_count--;
             scene_map[scene.index].Clear();
-            ReturnErrorOnFailure(this->Save(storage));
+            err = this->Save(storage);
+
+            if (err != CHIP_NO_ERROR)
+            {
+                global_scene_count.count_value++;
```

Comment:
Not sure what we are doing here. You decrease the global_scene_cout at line 443. then increase it back here (line 42) and there if scene Delete fails at line 456, we increment it back again at line 461.

Is this really the flow you wanted to implement or is there an operation error? Nonetheless, there are 3 saves of `global_scene_count` in quick succession. This should probably be optimized to only save it once

Reply:
This was requested as a way to ensure the Global Count/Scene Map/Scenes are matching in count in the event of a failure on any of the savings. If the scene map or scene save fails, changes to the global scene count must be undone. A return was missing from all the scenarios where insertion/delete, so there should only be one -- per ++ and vice versa now.